src/plugins/intel_gpu/tests/test_cases/experimental_detectron_generate_proposals_single_image_gpu_test.cpp

```diff
@@ -0,0 +1,211 @@
+
+        const primitive_id edgpsi_id = "experimental_detectron_generate_proposals_single_image";
+        const auto edgpsi_primitive = experimental_detectron_generate_proposals_single_image{edgpsi_id,
+                                                                                             input_im_info_id,
+                                                                                             input_anchors_id,
+                                                                                             input_deltas_id,
+                                                                                             input_scores_id,
+                                                                                             output_roi_scores_id,
+                                                                                             param.min_size,
+                                                                                             param.nms_threshold,
+                                                                                             param.pre_nms_count,
+                                                                                             param.post_nms_count};
+        topology.add(edgpsi_primitive);
+
+        network network(engine, topology);
+
+        network.set_input_data(input_im_info_id, input_im_info);
+        network.set_input_data(input_anchors_id, input_anchors);
+        network.set_input_data(input_deltas_id, input_deltas);
+        network.set_input_data(input_scores_id, input_scores);
+
+        const auto outputs = network.execute();
+
+        const auto rois = outputs.at(edgpsi_id).get_memory();
+
+        const cldnn::mem_lock<float> rois_ptr(rois, get_test_stream());
+        ASSERT_EQ(rois_ptr.size(), param.post_nms_count * 4);
+
+        const cldnn::mem_lock<float> roi_scores_ptr(output_roi_scores, get_test_stream());
+        ASSERT_EQ(roi_scores_ptr.size(), param.post_nms_count);
+
+        const auto &expected_roi_scores = param.expected_roi_scores;
+        const auto &expected_rois = param.expected_rois;
+        for (size_t i = 0; i < param.post_nms_count; ++i) {
+            EXPECT_NEAR(expected_roi_scores[i], roi_scores_ptr[i], 0.001) << "i=" << i;
+
+            // order of proposals with zero scores is not guaranteed (to be precise,
+            // it is not guaranteed for any equal score values)
+            if (expected_roi_scores[i] != 0.0f) {
+                for (size_t coord = 0; coord < 4; ++coord) {
+                    const auto roi_idx = i * 4 + coord;
+                    EXPECT_NEAR(expected_rois[roi_idx], rois_ptr[roi_idx], 0.001) << "i=" << i << ", coord=" << coord;
+                }
+            }
+        }
+    }
+};
+
+TEST_P(experimental_detectron_generate_proposals_single_image_test, basic) {
+    ASSERT_NO_FATAL_FAILURE(test());
+}
+
+INSTANTIATE_TEST_SUITE_P(
+        experimental_detectron_generate_proposals_single_image_gpu_test,
```

Comment:
Could you please add test case for fp16?

Reply:
done

Reply:
Thanks!